src/main/java/org/opensearch/sdk/handlers/OpensearchRequestHandler.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.sdk.handlers;
+
+import org.opensearch.extensions.OpenSearchRequest;
+import org.opensearch.sdk.NamedWriteableRegistryAPI;
+import org.opensearch.sdk.ExtensionsRunner;
+import org.opensearch.transport.TransportResponse;
+
+/**
+ * This class handles the request from OpenSearch to a {@link ExtensionsRunner#startTransportService(TransportService transportService)} call.
+ */
+
+public class OpensearchRequestHandler {
+    private NamedWriteableRegistryAPI namedWriteableRegistryApi = new NamedWriteableRegistryAPI();
+
+    /**
+     * Handles a request from OpenSearch and invokes the extension point API corresponding with the request type
+     *
+     * @param request  The request to handle.
+     * @return A response to OpenSearch for the corresponding API
+     * @throws Exception if the corresponding handler for the request is not present
+     */
+    public TransportResponse handleOpenSearchRequest(OpenSearchRequest request) throws Exception {
```

Comment:
This should be a class for `NamedWriteableRegistry` and not `OpenSearchRequest`. Also, we don't need a switch case here. If/else can do the job.

Reply:
I think we can keep this switch case here, it will easier for others adding more case in the future.